### ISSUE TITLE
[IDEA] Autosave frequency throttle

### ISSUE THREAD

**Issue by simonbaird** [Jermolene/TiddlyWiki5#6979]
**Is your feature request related to a problem? Please describe.**

I suspect that some brand new TiddlyWiki users on https://tiddlyhost.com don't realize they need to hit the save button to save their changes. I'd like to enable autosave for that reason, but with a throttle so that it autosaves only when it's more than (say) two minutes since the last save.

**Describe the solution you'd like**

I'm imagining a new configuration option related to autosave, called something like "Minimum seconds between autosaves" or "Autosave throttle (minimum seconds between autosaves)". The value would be an integer number of seconds stored in a tiddler named `$:/config/AutoSavePeriod` or something like that.

The Autosave mechanism would change so if that value is present and non-zero it would not trigger an autosave until the time since the last save is more than the threshold. Of course multiple autosave triggers would need to not queue up, so when the time comes only one save event happens.

**Describe alternatives you've considered**

My current workaround is to disable autosave on Tiddlyhost since I don't want every single change to trigger a save. The reasoning there is that: a). I think it would create a significantly higher load on Tiddlyhost's server, and b). since Tiddlyhost does a full file save every time there's a wait between "Save started..." and "Save complete". This is a bad UX at best and at worst it could result in race condition problems where different save request are happening at the same time.

I think this idea would be generally useful, rather than being a feature just for Tiddlyhost, so that's why I'm suggesting it as a TiddlyWiki feature.

**Additional context**

There may be some overlap with #6684 here.

Actually I'm not 100% this feature doesn't exist already, so please let me know this can be done already, or if there's a well known plugin that can do it.

**Comment by simonbaird** [comment]
Hypothetically, if I did some work on implementing this feature, would it be considered for merge?

**Comment by Jermolene** [comment]
Hi @simonbaird sorry for the delay, that's a great idea, and a PR would be appreciated.

**Comment by simonbaird** [comment]
I noticed there's something on TiddlyTools that might be relevant here, see https://tiddlytools.com/#TiddlyTools%2FTime%2FAutoSaver .

**Comment by saqimtiaz** [comment]
I have taken a quick crack at an implementation in #7995

**Comment by simonbaird** [comment]
See also [this forum post](https://talk.tiddlywiki.org/t/making-autosave-smarter-testing-a-new-implementation/13099).

**Comment by twMat** [comment]
As outlined [here](https://talk.tiddlywiki.org/t/autosave-would-be-useful-if/13111), I want to add that the OP (just like the current implementation), would be more useful if autosave was triggered not from clicking the *Done* button but rather from "any change" to the wiki (i.e whatever it is that triggers the *save-wiki* button to turn red).

**Comment by pmario** [comment]
@twMat -- You mean the delay trigger should be reset by every change - right? Saving with every change will kill direct saving to a server, where saving can take up to 1 minute for big wikis and or crappy internet connection

**Comment by twMat** [comment]
@pmario 

> You mean the delay trigger should be reset by every change - right? Saving with every change will kill direct saving to a server, where saving can take up to 1 minute for big wikis and or crappy internet connection

Yes, I do mean that the delay should be reset for every change. 

If you're saying that if the "delay-duration" is so short that saving is reinitiated even before the previous save was finished, then this could be avoided by defining some "minimum delay-duration". But isn't this the same problem if a user manually clicks save repeatedly? How is that handled? (On Tiddlyhost it seems to work fine; at least the little popup-saving-indicators behave as one would expect from a salvo of save clicks.)
